### ISSUE TITLE
LC0091 is still very unstable

### ISSUE THREAD

**Issue by jwikman** [StefanMaron/BusinessCentral.LinterCop#1068]
The LC0091 rule is still very unstable, even after (or because of?) the last changes in the preview.

I've got a small library app that gets false positives on LC0091 now and then. If I recompile the app, there could be no LC0091, or a few others - without any code being changed. 🤔

Everything is completely translated into sv-SE.

Below are the outputs of 6 consecutive compilation of that app, without changing anything between the compilations:

```
Microsoft (R) AL Compiler version 16.0.22.52049
Copyright (C) Microsoft Corporation. All rights reserved

Compilation started for project 'Smart Toolbox' containing '79' files at '10:04:41.220'.

d:\VSCode\Git\NAB\Smart Toolbox\App\src\RunTimeManagement.Codeunit.al(164,9): info LC0091: CanOnlyRunPageOrQueryErr is missing a translation in languages: sv-SE

Compilation ended at '10:04:43.221'.

Success: The package is created.

Microsoft (R) AL Compiler version 16.0.22.52049
Copyright (C) Microsoft Corporation. All rights reserved

Compilation started for project 'Smart Toolbox' containing '79' files at '10:06:51.956'.


Compilation ended at '10:06:54.187'.

Success: The package is created.

Microsoft (R) AL Compiler version 16.0.22.52049
Copyright (C) Microsoft Corporation. All rights reserved

Compilation started for project 'Smart Toolbox' containing '79' files at '10:06:59.381'.

d:\VSCode\Git\NAB\Smart Toolbox\App\src\RunTimeManagement.Codeunit.al(15,9): info LC0091: OnlyDeleteConnectedSessionsErr is missing a translation in languages: sv-SE
d:\VSCode\Git\NAB\Smart Toolbox\App\src\RunTimeManagement.Codeunit.al(16,9): info LC0091: ManuallyStoppedMsg is missing a translation in languages: sv-SE
d:\VSCode\Git\NAB\Smart Toolbox\App\src\RunTimeManagement.Codeunit.al(164,9): info LC0091: CanOnlyRunPageOrQueryErr is missing a translation in languages: sv-SE

Compilation ended at '10:07:01.063'.

Success: The package is created.

Microsoft (R) AL Compiler version 16.0.22.52049
Copyright (C) Microsoft Corporation. All rights reserved

Compilation started for project 'Smart Toolbox' containing '79' files at '10:07:05.637'.

d:\VSCode\Git\NAB\Smart Toolbox\App\src\TextFileImport\TextFileExporterImpl.Codeunit.al(233,9): info LC0091: DataTypeExcelErr is missing a translation in languages: sv-SE

Compilation ended at '10:07:07.440'.

Success: The package is created.


Microsoft (R) AL Compiler version 16.0.22.52049
Copyright (C) Microsoft Corporation. All rights reserved

Compilation started for project 'Smart Toolbox' containing '79' files at '10:07:13.726'.

d:\VSCode\Git\NAB\Smart Toolbox\App\src\TextFileImport\FileSettings.Table.al(19,13): info LC0091: Caption is missing a translation in languages: sv-SE

Compilation ended at '10:07:15.280'.

Success: The package is created.

Microsoft (R) AL Compiler version 16.0.22.52049
Copyright (C) Microsoft Corporation. All rights reserved

Compilation started for project 'Smart Toolbox' containing '79' files at '10:07:21.068'.

d:\VSCode\Git\NAB\Smart Toolbox\App\src\RunTimeManagement.Codeunit.al(164,9): info LC0091: CanOnlyRunPageOrQueryErr is missing a translation in languages: sv-SE

Compilation ended at '10:07:22.741'.

Success: The package is created.
```

How is this even possible? 😅

**Comment by jwikman** [comment]
Maybe this issue should be directed to @ChristianHovenbitzer? 🙂

**Comment by ChristianHovenbitzer** [comment]
@jwikman All issues regarding LC0091 to me 😅. I am already aware of this problem I just need to find some time to look into this. I was also able to reproduce the issue. 
Not sure how this happens though, my best guess is that there's an issue with the caching of nodes in the xliff files.

**Comment by jwikman** [comment]
I'm glad that you can repro yourself, it's a huge step in the direction of solving the issue! 🙂

**Comment by Arthurvdv** [comment]
@jwikman just a hunch here; Does it make any difference if you disable the parallellisme? You can set the with the `al.compilationOptions` in the settings.json file like this.

```json
"al.compilationOptions": {
  "parallel": false
}
```

**Comment by jwikman** [comment]
> [@jwikman](https://github.com/jwikman) just a hunch here; Does it make any difference if you disable the parallellisme? You can set the with the `al.compilationOptions` in the settings.json file like this.
> 
> "al.compilationOptions": {
>   "parallel": false
> }

Yes, it seems as you are spot on!

After disabling parallel compilation, I had no warning after 10-20 compilations. Enabling the setting gives me warnings about 4 out of 10 times again.

If this is the cause, do you have a theory on how to resolve the issue?

**Comment by Arthurvdv** [comment]
@ChristianHovenbitzer I had a similar problem during the development of Rule0089, so I've taken the liberty to make some changes to the rule which should resolve this issue I believe.

Can you have a look at https://github.com/StefanMaron/BusinessCentral.LinterCop/pull/1069 and let me know if it's okay to merge?

**Comment by ChristianHovenbitzer** [comment]
@Arthurvdv I added just one small detail

**Comment by Arthurvdv** [comment]
@ChristianHovenbitzer I can't seems to find anything you've added or changed?

**Comment by ChristianHovenbitzer** [comment]
@Arthurvdv I just commented without changing anything. I just wonderd because in your PR the action is registered on the `compilationStartContext` and I am not sure if the rule then also triggers without compiling

**Comment by Arthurvdv** [comment]
@ChristianHovenbitzer Your comment on the PR isn't visibele for me somehow 🤔 But valid question indeed, when I change a line of code in VS Code, without explicit starting a compile command, the `RegisterCompilationStartAction` trigger is executed.

Note that editing only the translation itself will not trigger anything, as the XLIFF files are not part of the compilation process of VS Code.

**Comment by ChristianHovenbitzer** [comment]
@Arthurvdv I think we should be good then, there was no update on xliff files previously as well.
